number_parser/parser.py

```diff
@@ -299,40 +303,33 @@ def parse_fraction(input_string, language=None):
     return None
 
 
-def parse_roman(input_string):
-    tokens = _tokenize(input_string, None)
-
-    def build_roman(roman_number):
-        roman = {'i': 1, 'v': 5, 'x': 10, 'l': 50, 'c': 100, 'd': 500, 'm': 1000}
-        num_tokens = re.split("^(m{0,3})(cm|cd|d?c{0,3})(xc|xl|l?x{0,3})(ix|iv|v?i{0,3})$", roman_number.lower())
-        num_tokens = [item for item in num_tokens if item != '']
-        built_num = 0
-        for num_token in num_tokens:
-            if re.search('iv|ix|xl|xc|cd|cm', num_token):
-                built_num += roman[num_token[1]] - roman[num_token[0]]
-            elif re.search('[vld][ixc]{1,3}', num_token):
-                built_num += roman[num_token[0]] + (roman[num_token[1]] * (len(num_token) - 1))
-            else:
-                built_num += roman[num_token[0]] * len(num_token)
-
-        return built_num
-
-    for token in tokens:
-        if re.search("^(m{0,3})(cm|cd|d?c{0,3})(xc|xl|l?x{0,3})(ix|iv|v?i{0,3})$", token.lower()):
-            tokens[tokens.index(token)] = str(build_roman(token))
-        final_sentence = ''.join(tokens)
-
-    return final_sentence
-
-
-def parse(input_string, language=None):
+def parse(input_string, language=None, numeral_systems=None):
     """
     Converts all the numbers in a sentence written in natural language to their numeric type while keeping
     the other words unchanged. Returns the transformed string.
     """
+    global complete_sentence
+
+    if numeral_systems is None:
+        numeral_systems = ['decimal', 'roman']
+
     if language is None:
         language = _valid_tokens_by_language(input_string)
 
+    for numeral_system in numeral_systems:
+
+        if numeral_system == 'decimal':
+            complete_sentence = _parse_decimal(input_string, language)
+            input_string = complete_sentence
+
+        if numeral_system == 'roman':
+            complete_sentence = _parse_roman(input_string)
+            input_string = complete_sentence
```

Comment:
You could do `result = input_string` before the loop, and then stop using `input_string` (it can be confusing for it to contain something other than the actual input string) and simplify those pairs of calls to e.g. `result = _parse_roman(result)`.

You could also use `elif`, and you could include an `else` to raise an exception if an input numeral system is unknown.